timescaledb/how-to-guides/skipscan.md

```diff
@@ -0,0 +1,83 @@
+TimescaleDB distributed hypertables. SkipScan is included in TimescaleDB 2.2.1
+and above.
+
+## About SkipScan
+When you query your database to find the most recent value of an item, you
+usually use a `DISTINCT` query. For example, you might want to find the latest
+stock or cryptocurrency price for each of your investments. Or you might have graphs
+and alarms that repeatedly query the most recent values for every device or
+service.
+
+As your tables get larger, `DISTINCT` queries tend to get slower. This is
+because PostgreSQL does not currently have a good mechanism for pulling a list
+of unique values from an ordered index. Even when you have an index that matches
+the exact order and columns for these kinds of queries, PostgreSQL scans the
+entire index to find all unique values. As a table grows, this operation keeps
+getting slower.
+
+SkipScan allows queries to incrementally jump from one ordered value to the next
+without reading all of the rows in between. Without support for this feature,
+the database engine has to scan the entire ordered index and then de-duplicate
+at the end, which is a much slower process.
+
+<highlight type="note">
+PostgreSQL has plans to implement a native feature like SkipScan, but it is
+unlikely to be included until at least PostgreSQL&nbsp;15. This section
+documents TimescaleDB SkipScan, which is not a native PostgreSQL feature.
+</highlight>
+
+SkipScan is an optimization for queries of the form `SELECT DISTINCT ON
+column_name`. Conceptually, SkipScan is a regular IndexScan that skips across an
+index looking for the next value that is greater than the current value.
+
+When you use a query with SkipScan, the `EXPLAIN` output includes a new node
+that can quickly return distinct items from a properly ordered index. With an
+IndexOnly scan, you have to scan the entire index, but SkipScan incrementally
+searches for each successive item in the ordered index. As it locates one item,
+the SkipScan node quickly restarts the search for the next item. This is a much
+more efficient way of finding distinct items in an ordered index.
+
+For benchmarking information on how SkipScan compares to regular `DISTINCT`
+queries, see our [SkipScan blog post][blog-skipscan].
+
+## Use SkipScan queries
+SkipScan is included in TimescaleDB 2.2.1 and above. This section describes how
+to set up your database index and query to use a SkipScan node.
+
+Your index must:
+* Contain the `DISTINCT` column as the first column.
+* Be a `BTREE` index.
+* Match the `ORDER BY` used in your query.
+
+Your query must:
+* Use the `DISTINCT` keyword on a single column.
+
+If the `DISTINCT` column is not the first column of the index, ensure any
+leading columns are used as constraints in your query. This means that if you
+are asking a question such as "retrieve a list of unique IDs in order" and
+"retrieve the last reading of each ID," you need at least one index like this:
```

Comment:
What you're describing isn't quite the same thing given the index below. I'd personally save this for later and just jump to the main use case for TimescaleDB users.

Most recent X ordered by time DESC.

What this paragraph is describing is when we might be searching for all "things" by a certain customer, and then we want to get the most recent reading by "thing". In that case you'd an index like this would still be used, first to filter on "customer" and then to SkipScan over the remaining portion of the index for that customer. (correct me if I'm wrong @svenklemm )

`CREATE INDEX "cpu_customer_tags_id_time_idx" ON readings (customer_id, tags_id, time DESC)`